### PR TITLE
Add CONTRIBUTING.md and update guides

### PR DESCRIPTION
**Pull request by vakrilov**
In this PR:
- Remove old an irrelevant articles
- Updated existing articles
- Added `Contributing.md` guide
- Added `WritingUnitTests.md` guide

CC @tjvantoll @jlooper

### CODE REVIEW

File: WritingUnitTests.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+```typescript
+import * as httpTests from "./http/http-tests";
+allTests["HTTP"] = httpTests;
+```
+
+## Writing Test Module
+The test modules are actually TypeScript modules which export unit tests and hooks as functions following this convention:
+
+* All exported function which begin with `test` prefix are unit-tests.
```
Comment (tjvantoll) [inline WritingUnitTests.md]: function --> functions

Comment (tjvantoll) [inline WritingUnitTests.md]: with a `test` prefix

File: WritingUnitTests.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+* The `setUpModule()` hook is called once - before all the tests in the module.
+* The `setUp()` hook is called before each tests.
+* The `tearDown()` hook called after each tests.
+* The `tearDownModule()` hook is called once - after all the tests in the module.
+
+# Asserting
+Test will fail if assert is not satisfied or if error is thrown during execution.
```
Comment (tjvantoll) [inline WritingUnitTests.md]: A test will fail if assert is not satisfied or if an error is thrown during execution.

File: WritingUnitTests.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    
+    TKUnit.assertNotNull(arr, "Array should be defined");
+    TKUnit.assertTrue(arr[2] > 2, "arr[2] is not big enough")
+    TKUnit.assertEqual(arr.length, 3, "Array length should be 3");
+}
+```
+
+# Async Tests
+
+Unit test can accept a single argument - a done callback. Test framework will wait for the `done()` callback to be called (or the test to timeout) before moving on.
```
Comment (tjvantoll) [inline WritingUnitTests.md]: Unit test --> Unit tests

The framework --> The test framework

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
+# Contributing to NativeScript
```
Comment (tjvantoll) [inline CONTRIBUTING.md]: Extremely trivial note: when this PR lands let me know I need to update a link I have to the current `.github/CONTRIBUTING.md` path on https://www.nativescript.org/contribute.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+```
+
+* Create a branch for your PR
+```bash
+git checkout -b my-fix-branch master
+```
+
+* The fun part! Make your code changes
+
+* Follow the [code conventions guide](CodeConvention.md)
```
Comment (tjvantoll) [inline CONTRIBUTING.md]: Suggestion: maybe add numbers to these requirements to make these next few sentences a little easier to read and follow. Here’s a rough idea of what I’m thinking.

* 1) Ensure your code follows the NativeScript code conventions guide.
* 2) Write unit tests for your fix or feature...
* 3) Rebase your changes to the latest version of the master branch of this repository.
* 4) Make sure all unit tests are green for iOS and Android.
* 5) Ensure your changes pass tslint validation.

When you’ve verified these changes to ahead and submit a PR...

File: WritingUnitTests.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
+# Writing Unit Tests for NativeScript Core Modules
+
+Unit tests for NativeScript Modules are written and executed with a custom lightweight test-runner and assertion framework.
+The purpose of this document is to get you familiar with it, so that you can unit-test your contributions to the NativeScript framework.
+
+# Run Unit Tests Project
+
+Refer to the [development-workflow guide](DevelopmentWorkflow.md) for instruction on how to setup your repo ready for development.
```
Comment (vakrilov) [inline WritingUnitTests.md]: That was useful! I guess I never really gave it a thought :)

**Pull request merged**